docs/src/main/sphinx/connector/delta-lake.rst

```diff
@@ -136,10 +136,11 @@ connector.
       - Description
       - Default
     * - ``delta.domain-compaction-threshold``
-      - Sets the number of transactions to act as threshold. Once reached the
-        connector initiates compaction of the underlying files and the delta
-        files. A higher compaction threshold means reading less data from the
-        underlying data source, but a higher memory and network consumption.
+      - Minimum size of query predicates above which Trino starts compaction.
+        Some databases perform poorly when a large list of predicates is pushed
```

Comment:
We should make this explanation about Delta Lake and not "some databases". Also with Delta Lake .. what does predicate pushdown even mean with Delta Lake .. 

we should link to https://trino.io/docs/current/optimizer/pushdown.html#predicate-pushdown and maybe expand that explanation so it covers what happens in an object storage system .. e.g. is there any processing happening in Delta Lake .. I dont think so ..

Reply:
With Delta Lake/Hive/Iceberg predicate pushdown means using predicates inside ORC and parquet readers to eliminate portions of the data based on min/max indexes maintained in those files. It is a processing of metadata that takes place in the connector code running on Trino workers.

Reply:
> We should make this explanation about Delta Lake and not "some databases". Also with Delta Lake .. what does predicate pushdown even mean

@mosabua I had edited the suggested description from @hashhar above https://github.com/trinodb/trino/pull/12965#discussion_r905746778. Will revise again.
> 
> we should link to https://trino.io/docs/current/optimizer/pushdown.html#predicate-pushdown and maybe expand that explanation so it covers what happens in an object storage system .. e.g. is there any processing happening in Delta Lake .. I dont think so ..

Based on @raunaqmorarka 's comment after yours, processing is not happening in Delta Lake but rather the connector code. His description was deemed as too specific by @findepi so I'm trying to craft a suitable description.